### PR TITLE
[bugfix] Fix causal attention mask for Blackwell FP4 MMA column layout

### PR DESCRIPTION
**Pull request by kevin314**
## Purpose

#1493 fixed non-causal attention by removing an incorrect V-row permutation in scaled_fp4_quant_trans_kernel. Causal attention remained broken due to a separate issue in the mask logic, which is addressed through this PR.

## Changes

The Blackwell FP4 MMA accumulator stores its N-dimension output with bits 1↔3 and 2↔4 swapped within each 32-column block relative to natural matrix column order. The causal mask in mainloop_tma_ws.h was comparing raw column indices from partition_C of the identity tensor directly against mask boundaries without undoing this permutation, causing wrong positions to be masked to -inf.

The fix is to add an actual_col lambda that undoes the MMA column permutation before any mask comparison:

```
auto actual_col = [](int c) {
    int local = c & 31;
    return (c & ~31) | (local & 1) | ((local & 24) >> 2) | ((local & 6) << 2);
};
```

Also removed the `masking_step > 0 guard on softmax_fused.rescale_o`, which was incorrectly skipping the output rescale on the first masking iteration.

```
Accuracy sweep — SDPA reference:
  [non-causal] B=2 H=4 L=128 D=64 — cos_sim=0.981914
  [non-causal] B=2 H=4 L=128 D=128 — cos_sim=0.982216
  [non-causal] B=2 H=4 L=256 D=64 — cos_sim=0.981776
  [non-causal] B=2 H=4 L=256 D=128 — cos_sim=0.982060
  [non-causal] B=2 H=4 L=512 D=64 — cos_sim=0.981425
  [non-causal] B=2 H=4 L=512 D=128 — cos_sim=0.981921
  [non-causal] B=2 H=4 L=1024 D=64 — cos_sim=0.981429
  [non-causal] B=2 H=4 L=1024 D=128 — cos_sim=0.981684
  [non-causal] B=2 H=4 L=2048 D=64 — cos_sim=0.980905
  [non-causal] B=2 H=4 L=2048 D=128 — cos_sim=0.982081
  [non-causal] B=2 H=4 L=4096 D=64 — cos_sim=0.981517
  [non-causal] B=2 H=4 L=4096 D=128 — cos_sim=0.981837
  [non-causal] B=1 H=4 L=128 D=128 — cos_sim=0.982467
  [non-causal] B=2 H=8 L=256 D=128 — cos_sim=0.982017
  [non-causal] B=1 H=4 L=384 D=128 — cos_sim=0.981758
  [causal] B=2 H=4 L=128 D=64 — cos_sim=0.985981
  [causal] B=2 H=4 L=128 D=128 — cos_sim=0.986065
  [causal] B=2 H=4 L=256 D=64 — cos_sim=0.985610
  [causal] B=2 H=4 L=256 D=128 — cos_sim=0.985546
  [causal] B=2 H=4 L=512 D=64 — cos_sim=0.984660
  [causal] B=2 H=4 L=512 D=128 — cos_sim=0.985709
  [causal] B=2 H=4 L=1024 D=64 — cos_sim=0.984427
  [causal] B=2 H=4 L=1024 D=128 — cos_sim=0.984969
  [causal] B=2 H=4 L=2048 D=64 — cos_sim=0.984374
  [causal] B=2 H=4 L=2048 D=128 — cos_sim=0.984268
  [causal] B=2 H=4 L=4096 D=64 — cos_sim=0.984169
  [causal] B=2 H=4 L=4096 D=128 — cos_sim=0.984206
  [causal] B=1 H=4 L=128 D=128 — cos_sim=0.986197
  [causal] B=2 H=8 L=256 D=128 — cos_sim=0.985849
  [causal] B=1 H=4 L=384 D=128 — cos_sim=0.985029

All tests completed.

Benchmark (B=2, H=32, bf16):
 B  H     L    D causal        ms   TFLOPS
--------------------------------------------------
 2 32   512   64  False     0.045    95.41
 2 32   512  128  False     0.044   193.73
 2 32  1024   64  False     0.052   330.02
 2 32  1024  128  False     0.079   437.42
 2 32  2048   64  False     0.141   487.37
 2 32  2048  128  False     0.259   529.92
 2 32  4096   64  False     0.523   525.36
 2 32  4096  128  False     0.821   669.54
 2 32   512   64   True     0.044    96.72
 2 32   512  128   True     0.044   193.56
 2 32  1024   64   True     0.045   382.84
 2 32  1024  128   True     0.062   555.48
 2 32  2048   64   True     0.103   670.25
 2 32  2048  128   True     0.181   758.90
 2 32  4096   64   True     0.365   752.11
 2 32  4096  128   True     0.512  1073.05

```
## Test Plan

Added fastvideo-kernel/tests/test_attn_qat_infer.py, a pytest suite comparing sageattn_blackwell against PyTorch's float32 SDPA reference across shapes (L∈{128…4096}, D∈{64,128}) in both causal and non-causal modes.

```bash
python fastvideo-kernel/tests/test_attn_qat_infer.py
```

## Test Results

<details>
<summary>Test output</summary>

```
Accuracy sweep — SDPA reference:
  [non-causal] B=2 H=4 L=128 D=64 — cos_sim=0.981914
  [non-causal] B=2 H=4 L=128 D=128 — cos_sim=0.982216
  [non-causal] B=2 H=4 L=256 D=64 — cos_sim=0.981776
  [non-causal] B=2 H=4 L=256 D=128 — cos_sim=0.982060
  [non-causal] B=2 H=4 L=512 D=64 — cos_sim=0.981425
  [non-causal] B=2 H=4 L=512 D=128 — cos_sim=0.981921
  [non-causal] B=2 H=4 L=1024 D=64 — cos_sim=0.981429
  [non-causal] B=2 H=4 L=1024 D=128 — cos_sim=0.981684
  [non-causal] B=2 H=4 L=2048 D=64 — cos_sim=0.980905
  [non-causal] B=2 H=4 L=2048 D=128 — cos_sim=0.982081
  [non-causal] B=2 H=4 L=4096 D=64 — cos_sim=0.981517
  [non-causal] B=2 H=4 L=4096 D=128 — cos_sim=0.981837
  [non-causal] B=1 H=4 L=128 D=128 — cos_sim=0.982467
  [non-causal] B=2 H=8 L=256 D=128 — cos_sim=0.982017
  [non-causal] B=1 H=4 L=384 D=128 — cos_sim=0.981758
  [causal] B=2 H=4 L=128 D=64 — cos_sim=0.985981
  [causal] B=2 H=4 L=128 D=128 — cos_sim=0.986065
  [causal] B=2 H=4 L=256 D=64 — cos_sim=0.985610
  [causal] B=2 H=4 L=256 D=128 — cos_sim=0.985546
  [causal] B=2 H=4 L=512 D=64 — cos_sim=0.984660
  [causal] B=2 H=4 L=512 D=128 — cos_sim=0.985709
  [causal] B=2 H=4 L=1024 D=64 — cos_sim=0.984427
  [causal] B=2 H=4 L=1024 D=128 — cos_sim=0.984969
  [causal] B=2 H=4 L=2048 D=64 — cos_sim=0.984374
  [causal] B=2 H=4 L=2048 D=128 — cos_sim=0.984268
  [causal] B=2 H=4 L=4096 D=64 — cos_sim=0.984169
  [causal] B=2 H=4 L=4096 D=128 — cos_sim=0.984206
  [causal] B=1 H=4 L=128 D=128 — cos_sim=0.986197
  [causal] B=2 H=8 L=256 D=128 — cos_sim=0.985849
  [causal] B=1 H=4 L=384 D=128 — cos_sim=0.985029

All tests completed.

Benchmark (B=2, H=32, bf16):
 B  H     L    D causal        ms   TFLOPS
--------------------------------------------------
 2 32   512   64  False     0.045    95.41
 2 32   512  128  False     0.044   193.73
 2 32  1024   64  False     0.052   330.02
 2 32  1024  128  False     0.079   437.42
 2 32  2048   64  False     0.141   487.37
 2 32  2048  128  False     0.259   529.92
 2 32  4096   64  False     0.523   525.36
 2 32  4096  128  False     0.821   669.54
 2 32   512   64   True     0.044    96.72
 2 32   512  128   True     0.044   193.56
 2 32  1024   64   True     0.045   382.84
 2 32  1024  128   True     0.062   555.48
 2 32  2048   64   True     0.103   670.25
 2 32  2048  128   True     0.181   758.90
 2 32  4096   64   True     0.365   752.11
 2 32  4096  128   True     0.512  1073.05
```

</details>

## Checklist

- [x] I ran `pre-commit run --all-files` and fixed all issues
- [x] I added or updated tests for my changes
- [ ] I updated documentation if needed
- [ ] I considered GPU memory impact of my changes

**For model/pipeline changes, also check:**
- [ ] I verified SSIM regression tests pass
- [ ] I updated the support matrix if adding a new model

### CODE REVIEW

**Comment by SolitaryThinker** [comment]
hi @Edenzzzz ptal, thanks!

File: fastvideo-kernel/tests/test_attn_qat_infer.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    start = torch.cuda.Event(enable_timing=True)
+    end = torch.cuda.Event(enable_timing=True)
+    start.record()
+    for _ in range(iters):
+        sageattn_blackwell(q, k, v, is_causal=is_causal)
+    end.record()
+    torch.cuda.synchronize()
+
+    ms = start.elapsed_time(end) / iters
+    flops = 4 * B * H * L * L * D / 1e12
```
Comment (Edenzzzz) [inline fastvideo-kernel/tests/test_attn_qat_infer.py]: flops /= 2 for causal
